src/server.coffee

```diff
@@ -260,7 +264,58 @@ startAuditUDPServer = (auditUDPPort, bindAddress) ->
 
   auditUDPServer.bind auditUDPPort, bindAddress
 
-  return defer
+  return defer.promise
+
+# function to start the TCP/TLS Audit server
+startAuditTcpTlsServer = (type, auditPort, bindAddress) ->
+  defer = Q.defer()
+
+  # data handler
+  handler = (sock) ->
+    message = ""
+    length = 0
+
+    sock.on 'data', (data) ->
+      # convert to string and concatenate
+      message += data.toString()
+
+      # check if length is is still zero and first occurannce of space
+      if length == 0 and message.indexOf(' ') != -1
+        # get index of end of message length
+        lengthIndex = message.indexOf " "
+
+        # source message length
+        lengthValue = message.substr(0, lengthIndex)
+
+        # remove white spaces
+        length = parseInt(lengthValue.trim())
+        
+        # update message to remove length - add one extra character to remove the space
+        message = message.substr(lengthIndex + 1)
+      
+      # if sourced length equals message length then full message received
+      if length == message.length
+        logger.info "[Auditing #{type}] Received message from #{sock.remoteAddress}"
+        auditing.processAudit message, -> logger.info "[Auditing #{type}] Processed audit"
```

Comment:
Here you may want to reset `message` and `length` so that if you receive another audit on the same connection you are ready to process it.